test/src/unit-json_patch.cpp

```diff
@@ -1268,7 +1280,7 @@ TEST_CASE("JSON patch")
             std::ifstream f(filename);
             json suite = json::parse(f);
 
-            for (const auto& test : suite)
+            for (const auto test : suite)
```

Comment:
The tests fail with a SIGSEGV if we process `suite` by a const reference. I have no idea why...

Reply:
There is probably an error in the move constructor of `basic_json`. When object is moved, its children still have old (no longer existing) parent. The following may help:
```
$ git diff include/nlohmann/json.hpp
diff --git a/include/nlohmann/json.hpp b/include/nlohmann/json.hpp
index 2fc497bae..f6e19eead 100644
--- a/include/nlohmann/json.hpp
+++ b/include/nlohmann/json.hpp
@@ -2196,6 +2196,10 @@ class basic_json
         other.m_type = value_t::null;
         other.m_value = {};

+#if JSON_DIAGNOSTICS
+        for (auto& item : *this) { item.m_parent = this; }
+#endif
+
         assert_invariant();
     }
#endif
```
The copy constructor has more elaborated logic for the same purpose. By the way, the copy constructor initializes new object with no parent - i. e. with no danging pointer (if there is one). This is why processing `suite` by value works.

Reply:
Thanks a lot! I missed that one. Fixed in 294fa343d54634856dfef4a645de5241bd2dcde1.